operator/redisfailover/service/client.go

```diff
@@ -230,6 +202,50 @@ func (r *RedisFailoverKubeClient) DestroydOrphanedRedisNetworkPolicy(rf *redisfa
 	return err
 }
 
+func (r *RedisFailoverKubeClient) DestroyOrphanedRedisSlaveHaProxy(rf *redisfailoverv1.RedisFailover) error {
+
+	// Helper function to handle the deletion of resources
+	deleteResource := func(namespace, name string, getter func(namespace, name string) (interface{}, error), deleter func(namespace, name string) error) error {
+		_, err := getter(namespace, name)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return nil
+			}
+			return err
+		}
+		return deleter(namespace, name)
+	}
+
+	resourceTypes := map[string]struct {
+		getter  func(namespace, name string) (interface{}, error)
+		deleter func(namespace, name string) error
+	}{
+		"service": {
+			getter:  func(namespace, name string) (interface{}, error) { return r.K8SService.GetService(namespace, name) },
+			deleter: r.K8SService.DeleteService,
+		},
+		"configmap": {
+			getter:  func(namespace, name string) (interface{}, error) { return r.K8SService.GetConfigMap(namespace, name) },
+			deleter: r.K8SService.DeleteConfigMap,
+		},
+		"deployment": {
+			getter:  func(namespace, name string) (interface{}, error) { return r.K8SService.GetDeployment(namespace, name) },
+			deleter: r.K8SService.DeleteDeployment,
+		},
+	}
+
+	name := GetHaproxySlaveName(rf)
+
+	for _, resType := range []string{"service", "configmap", "deployment"} {
+		resource := resourceTypes[resType]
+		if err := deleteResource(rf.Namespace, name, resource.getter, resource.deleter); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
The operator uses a more explicit, easier to read style to clean up its sentinel resources:

https://github.com/powerhome/redis-operator/blob/bd0d4339ce467ddf1396ec8b76ebb48093b395e9/operator/redisfailover/service/client.go#L164-L187

Is there anything about removing the haroxy resources which necessitates the added complexity? This indirection adds quite a bit of cognitive overhead to the reader.

Reply:
The aim was not to repeat if/else condition,to DRY a bit code

Reply:
but we can simplify the code, no problems